app/Services/EmpresaService.php

```diff
@@ -25,17 +25,37 @@ public function getById($id){
         ->getById($id);
     }
 
-    public function register($user_id,$nome,$nome_social,$razao_social,$cnpj,$telefone,$email)
+    public function register($user_id,$nome,$nome_social,$razao_social,$cnpj,$telefone,$email,$tipo_empresa_id,$natureza_empresa_id,$inscricao_empresa_id)
     {
-        $result = $this->empresaRepository
-        ->save($user_id,$nome,$nome_social,$razao_social,$cnpj,$telefone,$email);
+        DB::beginTransaction();
+        try{
+        DB::commit();
```

Comment:
o DB::commit() deve ser depois que é montado o objeto, o commit ele commita as informações me banco, se for chamado antes de popular o objeto não faz o efeito esperado

Reply:
você fala do controller e services? 

se for eu tirei do services e deixei so no controller.

Reply:
ah ta , entendi.